questionary/prompt.py

```diff
@@ -166,6 +162,11 @@ def unsafe_prompt(
                 raise ValueError(
                     "'when' needs to be function that accepts a dict argument"
                 )
+
+        choices = question_config.get("choices")
+        if choices is not None and callable(choices):
+            question_config["choices"] = choices(answers)
```

Comment:
I think there might be an issue here - updating this in the `question_config` dict won't populate this change to the `_kwargs` variable with is used further down to create the question.